src/rgw/services/svc_bilog_rados.cc

```diff
@@ -18,15 +18,15 @@ void RGWSI_BILog_RADOS::init(RGWSI_BucketIndex_RADOS *bi_rados_svc)
   svc.bi = bi_rados_svc;
 }
 
-int RGWSI_BILog_RADOS::log_trim(const RGWBucketInfo& bucket_info, int shard_id, string& start_marker, string& end_marker)
+int RGWSI_BILog_RADOS::log_trim(const RGWBucketInfo& bucket_info, int shard_id, const bucket_log_layout_generation& log_layout, string& start_marker, string& end_marker)
 {
   RGWSI_RADOS::Pool index_pool;
   map<int, string> bucket_objs;
 
   BucketIndexShardsManager start_marker_mgr;
   BucketIndexShardsManager end_marker_mgr;
 
-  int r = svc.bi->open_bucket_index(bucket_info, shard_id, &index_pool, &bucket_objs, nullptr);
+  int r = svc.bi->open_bucket_index(bucket_info, shard_id, bucket_info.layout.current_index, &index_pool, &bucket_objs, nullptr);
```

Comment:
`log_trim()` is taking `bucket_log_layout_generation` as an argument. it should use the new `log_to_index_layout()` to convert that into the index layout we pass to `open_bucket_index()`

Reply:
thanks for catching this!